File: output/struct.go
package main

type RootSomeObjectSubobj struct {
	wha whaStruct
}

type RootSomeObject struct {
	key1 string
	key2 int
	subobj RootSomeObjectSubobj
}

type Root struct {
	someBoolean bool
	someInteger int
	someString string
	someBooleanEnum someBooleanEnumStruct
	someIntegerEnum someIntegerEnumStruct
	someStringEnum someStringEnumStruct
	someArrayOfBooleans []bool
	someArrayOfIntegers []int
	someArrayOfStrings []string
	someObject RootSomeObject
}

File: output/struct.ts
type RootSomeObjectSubobj = {
	wha: Array<'uwu' | 'owo' | 'rawr'>
}

type RootSomeObject = {
	key1: string
	key2: number
	subobj: RootSomeObjectSubobj
}

type Root = {
	someBoolean: boolean
	someInteger: number
	someString: string
	someBooleanEnum: Array<'true'>
	someIntegerEnum: Array<'0' | '3'>
	someStringEnum: Array<'a' | 'b' | 'c'>
	someArrayOfBooleans: Array<boolean>
	someArrayOfIntegers: Array<number>
	someArrayOfStrings: Array<string>
	someObject: RootSomeObject
}

